db/migrate/20161231115932_add_full_text_search_indexes.rb

```diff
@@ -0,0 +1,9 @@
+class AddFullTextSearchIndexes < ActiveRecord::Migration[5.0]
+  def up
+    execute "create index notifications_subject_title on notifications using gin(to_tsvector('english', subject_title))"
+  end
+
+  def down
+    execute "drop index notifications_subject_title"
+  end
+end
```

Comment:
I tried it on a bigger db and it still wasn't used, but after filtering by user_id it's really quick at the moment, let's remove this for now and revisit again if performance get's bad once the db is much bigger!